src/coreclr/jit/optcse.cpp

```diff
@@ -785,10 +785,12 @@ bool Compiler::optValnumCSE_Locate()
                 }
 
                 // Don't allow CSE of constants if it is disabled
-                //
                 if (tree->IsIntegralConst())
                 {
-                    if (!enableConstCSE)
+                    if (!enableConstCSE &&
```

Comment:
It is unsatisfying that we will now CSE things even when explicitly asked not to (`CONST_CSE_DISABLE_ALL`).

Perhaps the easiest solution would be to delete `CONST_CSE_DISABLE_ALL`. Or check it separately (`if (forceDisableConstCSE) continue;`).

Reply:
I agree, I'll delete CONST_CSE_DISABLE_ALL in a follow up PR, thanks